### PR TITLE
feat: Automatically check user answers

### LINKED ISSUE #41: Automatically check answers
The bot should have some way to automatically figure out what the correct answers are for a given day. Otherwise, the owner will have to do daily manual maintenance

### PR DESCRIPTION
**Pull request by proegssilb**
Any time a user submits code, check the answers against the AOC website if we don't already have them.

I do expect to have to rework this, but an attempt was made to make a reasonably robust solution, and also to do some clean-up before publishing.

Resolves #41 .

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ dynaconf = "^3.2.4"
 tzdata = "^2023.3"
 # this is used in fetch.py
 requests = "^2.31.0"
+aiohttp = "^3.9.1"
```
Comment (proegssilb) [inline pyproject.toml:17]: It looks like requests doesn't have async support built-in anymore, or I would have just used requests. Maybe I missed something.

File: lib.py [hunk 2]
```diff
@@ -303,6 +313,36 @@ def process_run_result(
...
+    async with session.post(answer_url, data=payload, cookies={"session": token}) as response:
+        response_text = await response.text()
+        if "That's not the right answer" in response_text:
+            return False
+        elif "That's the right answer" in response_text:
+            # We have a correct answer
+            answers_map[answer_key] = answer
+            return True
+        else:
+            logger.error("Unrecognized response from AOC website:\n%s", response_text)
```
Comment (proegssilb) [inline lib.py:343]: There's a pretty huge gap here: If an answer is queried, but doesn't make it to the DB, the website will no longer give us that answer via the URL we're using. Instead, we have to parse the raw problem page (eg, `https://adventofcode.com/2023/day/12`) in order to find the answers we've already unlocked. 

The work-around is easy enough, just manually load the appropriate answer into the DB.

We will almost certainly bump into this limitation eventually. The question is _how often_ we'll hit the issue.

File: lib.py [hunk 3]
```diff
@@ -47,21 +46,32 @@ async def benchmark(
...
+                            if await try_answer(
+                                session, day, part, session_label, result.answer, answers_map
+                            ):
+                                result.verified = True
+                                db.set_verified_answer(year(), day, part, in_file, str(result.answer))
+                            else:
+                                # TODO: Take more stringent action.
+                                result.verified = False
+
+                        if result is not None and result.median is not None and result.average is not None:
```
Comment (proegssilb) [inline lib.py:69]: The logic change here is to resolve an issue mypy pointed out.

File: pyproject.toml (hunk 1 above)
Comment (ultrabear) [reply pyproject.toml:17]: requests has never had async support afaik

**Comment by proegssilb** [comment]
These changes should probably just be re-written, and a new PR created.

**Pull request closed without merging**